Sources/ClientRuntime/EventStream/MessageDecoderStream.swift

```diff
@@ -6,5 +6,26 @@
 //
 
 /// Stream adapter that decodes input data into `EventStream.Message` objects.
-public protocol MessageDecoderStream: AsyncSequence {
+public protocol MessageDecoderStream: AsyncSequence where Event == Element {
+    associatedtype Event
+}
+
+extension MessageDecoderStream {
+    /// Returns an `AsyncThrowingStream` that decodes input data into `Event` objects.
+    public func toAsyncStream() -> AsyncThrowingStream<Event, Error> {
```

Comment:
Would we expect this function to ever be overridden by conforming types? 
If yes, can we also make this function part of the protocol conformance to avoid unexpected behavior. 

If `toAsyncStream` is not part of the protocol conformance and a conforming type overrides the function, the implementation that is called depends on the casted type of the instance.

Reply:
I don't expect it to be overridden because it is generic and works on the Element than any specific type.